example/appengine/app.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// The demo app shows how to use the github package on Google App Engine.
+package demo
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/log"
+)
+
+func init() {
+	http.HandleFunc("/", handler)
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	if r.URL.Path != "/" {
+		http.NotFound(w, r)
+		return
+	}
+
+	ctx := appengine.NewContext(r)
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: os.Getenv("GITHUB_AUTH_TOKEN")},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+
+	commits, _, err := client.Repositories.ListCommits(ctx, "google", "go-github", nil)
+	if err != nil {
+		log.Errorf(ctx, "ListCommits: %v", err)
+		return
```

Comment:
You haven't written anything to `w` yet, so you could also do `http.Error(w, err.Error(), http.Status...)` here.